### PR TITLE
Fix the protected config keys for HttpsSegmentFetcher

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Also auto-format the segment fetcher related classes

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HdfsSegmentFetcher.java [hunk 1]
```diff
@@ -78,15 +72,15 @@ private void authenticate(Configuration hadoopConf, org.apache.commons.configura
       UserGroupInformation.setConfiguration(hadoopConf);
       if (UserGroupInformation.isSecurityEnabled()) {
         try {
-          if (!UserGroupInformation.getCurrentUser().hasKerberosCredentials()
-              || !UserGroupInformation.getCurrentUser().getUserName().equals(principal)) {
+          if (!UserGroupInformation.getCurrentUser().hasKerberosCredentials() || !UserGroupInformation.getCurrentUser()
+              .getUserName()
```
[the remark this answers is not in the document]
Comment (mcvsubbu) [reply pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HdfsSegmentFetcher.java]: +1, please avoid mixing format related changes with logic changes. Make a separate PR for format related changes if they bother you.

File: pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/SegmentFetcher.java [hunk 2]
```diff
@@ -20,15 +20,15 @@
...
+
 public interface SegmentFetcher {
 
   void init(Configuration configs);
 
   void fetchSegmentToLocal(String uri, File tempFile) throws Exception;
 
   /**
-   * @return a list of config keys whose value should not be logged.
+   * Returns a list of config keys whose value should not be logged.
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/SegmentFetcher.java]: what's wrong with the javadoc convention? I prefer to have javadocs in interface classes

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/SegmentFetcher.java]: Me too. This is just putting it as description in javadoc. Will add back the *@return*

File: pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HdfsSegmentFetcher.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-common/src/main/java/com/linkedin/pinot/common/segment/fetcher/HdfsSegmentFetcher.java]: Reverted

**Pull request merged**